SessionMessagingKit/Shared Models/SessionThreadViewModel.swift

```diff
@@ -528,8 +551,9 @@ public extension SessionThreadViewModel {
                         \(interaction[.authorId]),
                         \(interaction[.linkPreviewUrl]),
             
-                        SUM(\(interaction[.wasRead]) = false) AS \(ViewModel.threadUnreadCountKey),
-                        SUM(\(interaction[.wasRead]) = false AND \(interaction[.hasMention]) = true) AS \(ViewModel.threadUnreadMentionCountKey)
+                        SUM(\(interaction[.wasRead]) = false AND \(interaction[.variant]) IN (\(Interaction.Variant.standardIncoming), \(Interaction.Variant.infoCall))) AS \(ViewModel.threadUnreadCountKey),
+                        SUM(\(interaction[.wasRead]) = false AND \(interaction[.hasMention]) = true) AS \(ViewModel.threadUnreadMentionCountKey),
+                        (SUM(\(interaction[.wasRead]) = false) > 0) AS \(ViewModel.threadHasUnreadMessagesOfAnyKindKey)
```

Comment:
So the `willInsert` and `canBeUnread` logic on an `Interaction` _should_ prevent unexpected interaction types from being classed as unread so we shouldn't really need additional checks here (unless we add a DB query which manually marks interactions as unread)

If you do want to include the `variant` checks in these queries we'll need to add a new index to the table to prevent this impacting performance

Reply:
This is similar to the use of `threadHasUnreadMessagesOfAnyKind`. In my understanding, we don't want those info messages to be count in the unread count, so I need to filter them out...

Reply:
No worries, responded above re. the `threadHasUnreadMessagesOfAnyKind` bit - should still look at adding a new index (adding the below to the `_011_DisappearingMessagesConfiguration` migration should do it):
```
db.create(
    index: "interaction_on_wasRead_and_variant",
    on: Interaction.databaseTableName,
    columns: [Interaction.Column.wasRead, Interaction.Column.variant].map { $0.name }
)
```

Reply:
Would also be good to update all of the logic to use the same `variantsToIncrementUnreadCount` for consistency (ie `SUM(\(interaction[.wasRead]) = false AND \(interaction[.variant]) IN \(Interaction.Variant.variantsToIncrementUnreadCount)) AS \(ViewModel.threadUnreadCountKey)`